packages/react-component-library/src/components/Timeline/Row.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import classNames from 'classnames'
+import { differenceInDays } from 'date-fns'
+
+import { getKey, isOdd } from './helpers'
+import { TimelineContext } from './context'
+import { TimelineEvent } from './context/types'
+import { DAY_WIDTH } from './constants'
+import { Event } from '.'
+
+export interface RowProps extends ComponentWithClass {
+  events: TimelineEvent[]
+}
+
+export const Row: React.FC<RowProps> = ({ events }) => {
+  return (
+    <div className="timeline__row">
+      <div className="timeline__row-weeks">
+        <TimelineContext.Consumer>
+          {({ state: { months, weeks } }) => {
+            return weeks.map(({ startDate }, index) => {
```

Comment:
I wonder if extract 1 or 2 functions would work well here just to help with readability.

Reply:
Potentially consider a `RowWeeks` component.